src/commonMain/kotlin/fr/acinq/lightning/channel/Channel.kt

```diff
@@ -283,6 +234,129 @@ sealed class ChannelState {
+                    val closingTx = newState.mutualClosePublished.first()
+                    val finalAmount = closingTx.toLocalOutput?.amount ?: 0.sat
+                    val address = closingTx.toLocalOutput?.publicKeyScript?.let {
+                        Helpers.Closing.btcAddressFromScriptPubKey(
+                            scriptPubKey = it,
+                            chainHash = staticParams.nodeParams.chainHash
+                        )
+                    } ?: "unknown"
+                    listOf(
+                        ChannelAction.Storage.StoreOutgoingPayment.ViaClose(
+                            amount = finalAmount,
+                            miningFees = channelBalance.truncateToSatoshi() - finalAmount,
+                            address = address,
+                            txId = closingTx.tx.txid,
+                            isSentToDefaultAddress = closingTx.toLocalOutput?.publicKeyScript == oldState.commitments.params.localParams.defaultFinalScriptPubKey,
+                            closingType = ChannelAction.Storage.StoreOutgoingPayment.ViaClose.Type.Mutual
+                        )
+                    )
+                }
+                else -> {
+                    // this is a force close, the closing tx is a commit tx
+                    // since force close scenarios may be complicated with multiple layers of transactions, we estimate global fees by listing all the final outputs
+                    // going to us, and subtracting that from the current balance
+                    val (commitTx, type, finalOutputs) = when {
+                        newState.localCommitPublished is LocalCommitPublished -> {
+                            val finalOutputs = buildList {
+                                newState.localCommitPublished.claimMainDelayedOutputTx?.let { addAll(it.tx.txOut) }
+                                addAll(newState.localCommitPublished.claimHtlcDelayedTxs.flatMap { it.tx.txOut })
+                            }
+                            Triple(newState.localCommitPublished.commitTx, ChannelAction.Storage.StoreOutgoingPayment.ViaClose.Type.Local, finalOutputs)
+                        }
+                        newState.remoteCommitPublished is RemoteCommitPublished -> {
+                            val finalOutputs = buildList {
+                                newState.remoteCommitPublished.claimMainOutputTx?.let { addAll(it.tx.txOut) }
+                                addAll(newState.remoteCommitPublished.claimHtlcTxs.values.filterNotNull().flatMap {
+                                    when (it) {
+                                        is Transactions.TransactionWithInputInfo.ClaimHtlcTx.ClaimHtlcSuccessTx -> it.tx.txOut
+                                        is Transactions.TransactionWithInputInfo.ClaimHtlcTx.ClaimHtlcTimeoutTx -> it.tx.txOut
+                                    }
+                                })
```

Comment:
We don't need the pattern matching, do we? Same for the other cases below.

```suggestion
                                addAll(newState.remoteCommitPublished.claimHtlcTxs.values.filterNotNull().flatMap { it.tx.txOut })
```

Reply:
Fixed in https://github.com/ACINQ/lightning-kmp/pull/458/commits/e47435021275f95ffbcf6a6a32e85405abf6b65b